### DISCUSSION TITLE
Making a FOSS version

### DISCUSSION THREAD

**Discussion by axx0** [axx0/Civ2-clone#102] [Ideas]
@crhy mentions the possibility of going from open-source to free open-source by making custom art and thus not being dependent on the original's assets to play the game.
The question is if we want to go in that direction (a lot of work would have to be put into replacing all the assets, including sound files, also there are lots of text files).
Or maybe give the player an option to use old or new art. If some of the new files are missing, the game automatically uses old files.

**Comment by reubene** [comment]
It's not a bad idea, the some parts of the existing image sets will need extending anyway as they only support 8 civs and the engine can support much more than that.

**Comment by crhy** [comment]
It seems like we could just use AI to make sprites of all the existing
units pretty quickly. I bet it could be done in less than 24 hours!

**Reply by crhy** [reply]
Also I would say skip all sound initially.  I usually play with the sound off anyway.  We can add sound back later.  I do have a studio though, and could record the sounds needed.

**Reply by crhy** [reply]
https://files.catbox.moe/dgzyc0.png

**Reply by crhy** [reply]
obviously we'd want a greenscreen or transparency background....

**Reply by crhy** [reply]
https://files.catbox.moe/u83ved.png

**Reply by crhy** [reply]
https://files.catbox.moe/44x3fd.png

**Reply by crhy** [reply]
How big should the sprites be?  Is 1024x1024 a good size?  Would be good to have them a little future proof....

**Reply by crhy** [reply]
https://files.catbox.moe/n0g626.png

**Reply by crhy** [reply]
https://files.catbox.moe/hsqgaz.png

**Reply by crhy** [reply]
https://files.catbox.moe/h7hawq.png

**Reply by crhy** [reply]
https://files.catbox.moe/mt2ria.png

**Reply by crhy** [reply]
https://files.catbox.moe/2zfake.png

**Reply by crhy** [reply]
https://files.catbox.moe/eh5365.png

**Reply by crhy** [reply]
https://files.catbox.moe/zbgj2j.png

**Reply by crhy** [reply]
This is really fun.  These are SO much better than the original art.  https://files.catbox.moe/kz684w.png

**Reply by crhy** [reply]
OMG This spy is the best one yet:  https://files.catbox.moe/69e6pr.png

**Reply by crhy** [reply]
https://playground.com/create is what I'm using if you guys want to join in.

**Comment by axx0** [comment]
> How big should the sprites be? Is 1024x1024 a good size? Would be good to have them a little future proof....

This all depends on the unit tile size you want to have. Right now the game has fixed unit sizes 64x48 (for MGE). We would have to make a logic that assumes different unit sizes (based on a parameter or read from the png itself) and render the background based on that. Also to consider is the game can zoom in x2.
Actually if we want to expand the game (not just sprites) we will have to came up with different rules, images, etc. logic so that we're not limited to the number of civs, units, improvements, etc. of the original. We're far off from that point but it doesn't hurt to start talking about it.

Anyway compliments on the art you've done. Especially the spy. I like how she's hiding her hands in the pocket, not exposing those 7 fingers.

**Reply by crhy** [reply]
So let's just stick with 1024x1024 for now.  It will be future proof for a long, long time.  This is a 2d game after all, and as the funnest game ever, I see no need to alter that yet!  Yes the spy is SO GOOD!  I'll make a few more now.  Should I upload them to a directory here on github, or.... ?

**Reply by axx0** [reply]
Yes, probably best to put it in a folder on github so it can be used later.

**Comment by crhy** [comment]
Ok, I'll do that.

**Reply by crhy** [reply]
I can't create files without forking the repo....  Can you just make a couple of folders, one for units, one for city improvements, one for wonders, one for terrain types, and one for Civilzation Advances?  I'll slowly fill the folders with all the art we need to go fully FOSS.

Let's go with 1024x1024 Transparent PNG files, and we'll see if .GIF or something is better later?  Or let me know now what you guys think.  
![two-napolean-era-dragoons-in-battle-887176028](https://github.com/axx0/Civ2-clone/assets/21183395/8d2905d6-fc4e-4380-a2da-800ecf0789f7)
![two-men-in-blue-uniforms-firing-rifles-in-the-1800s-235577188](https://github.com/axx0/Civ2-clone/assets/21183395/fa25605c-39e5-4a06-a83a-0e5ea02bc837)
![two-men-in-blue-uniforms-kneeling-and-firing-rifles-in-the-1800s-339333675](https://github.com/axx0/Civ2-clone/assets/21183395/80f8ab49-1060-4e0b-8a83-15bbaf55c5fc)
![paratrooper-actively-parachuting-274011233](https://github.com/axx0/Civ2-clone/assets/21183395/8ffbacbf-9cc6-45b0-8725-90cd41fbb414)
![two-angry-medeival-men-in-iron-helmets-holding-pikes-502227578](https://github.com/axx0/Civ2-clone/assets/21183395/310eeeb4-18f0-44e9-b6f1-5b545272a114)
![two-angry-partisans-holding-rifles-over-their-heads-wearing-black-uniforms-862951040](https://github.com/axx0/Civ2-clone/assets/21183395/0e02d918-5ed1-4e6c-984c-67d6881bd61e)
![a-ww2-german-submarine-on-the-open-ocean-from-above-36663140](https://github.com/axx0/Civ2-clone/assets/21183395/4d7db122-b4b8-4738-b182-a553fcf9021a)
![two-war-elephants-on-attack-40613638](https://github.com/axx0/Civ2-clone/assets/21183395/cca4ea69-b531-432e-94af-11dcb6c2dce0)
![two-battle-weary-marines-holding-their-weapons-from-above-168226193](https://github.com/axx0/Civ2-clone/assets/21183395/f29c3a78-210a-4dd1-bd4a-934f103a0e34)
![a-humvee-with-machine-gun-mounted-on-top-from-above-999311884](https://github.com/axx0/Civ2-clone/assets/21183395/367d5e7c-9eea-4b19-9b1e-ed088e29e0f4)
![a-humvee-with-machine-gun-mounted-from-above-25065260](https://github.com/axx0/Civ2-clone/assets/21183395/b5040e2f-777b-4a3a-83ad-4739e2ea16b4)
![a-1600s-galleon-from-above-with-sails-deployed-on-the-open-ocean-843972639](https://github.com/axx0/Civ2-clone/assets/21183395/735ab2f6-57f7-4da6-9dd9-3e7e3b463c53)
![a-1600s-galleon-from-above-with-sails-deployed-402904649](https://github.com/axx0/Civ2-clone/assets/21183395/dfc109eb-bbca-45f7-8931-b872801544e5)
![a-1700s-frigate-from-above-with-sails-deployed-979574648](https://github.com/axx0/Civ2-clone/assets/21183395/fdfd22e4-322a-47ef-a86f-f098563d2a84)
![a-frigate-from-above-335073257](https://github.com/axx0/Civ2-clone/assets/21183395/7d9e9fe6-7e98-44f2-aaaf-6d8b6086f76c)
![two-men-on-horses-carrying-spears-780223548](https://github.com/axx0/Civ2-clone/assets/21183395/76e0e1da-e048-44a9-8962-c901d89b2a54)
![apache-helicopter-271976486](https://github.com/axx0/Civ2-clone/assets/21183395/713da80f-deac-48ba-87eb-cf54ffa2801e)
![two-angry-men-in-crimson-robes-holding-torches](https://github.com/axx0/Civ2-clone/assets/21183395/1c50bacf-e55d-4ac2-aaff-d05fe2fe450d)
![two-angry-men-in-crimson-robes-holding-torches-460223480](https://github.com/axx0/Civ2-clone/assets/21183395/6248f0fb-4874-4ad5-ad2f-98036f6ae64c)
![two-men-of-different-ethnicity-in-loincloths-with-ancient-shovels-658531085](https://github.com/axx0/Civ2-clone/assets/21183395/222740ab-1f88-4ca4-b879-170ea879eb45)
![two-standing-mountain-men-in-coon-skin-caps-842153717](https://github.com/axx0/Civ2-clone/assets/21183395/97cd1df9-a56f-4b30-a964-ebcf0fc5a8ff)
![uss-troop-transport-world-war-2-683373936](https://github.com/axx0/Civ2-clone/assets/21183395/6339c37a-98bc-4e67-ac22-cc185d8c46e4)
![uss-cruiser-308610912](https://github.com/axx0/Civ2-clone/assets/21183395/b7d4a0fc-6f11-4c54-9d3b-c30b2f043279)
![uss-destroyer-579889387](https://github.com/axx0/Civ2-clone/assets/21183395/a9e0814d-3fcf-45ba-bb5a-ce36d5253441)
![bismark-battleship-468442743](https://github.com/axx0/Civ2-clone/assets/21183395/421e1d33-7375-4fca-8a41-c81782ea2e5c)
![uss-lexington-with-spitfire-flying-above-in-the-open-ocean-479989699](https://github.com/axx0/Civ2-clone/assets/21183395/b54bcfdf-abeb-4d1d-8a8a-2addf64c5389)
![civil-war-era-ironclad-606863511](https://github.com/axx0/Civ2-clone/assets/21183395/2659d634-67ab-4c61-b363-3241607960ec)
![ancient-single-square-sail-trireme-520218352](https://github.com/axx0/Civ2-clone/assets/21183395/6f530d14-3949-4d5a-8fe2-468a026f490f)
![trireme-sailing-523684153](https://github.com/axx0/Civ2-clone/assets/21183395/5d4ce2e8-d117-47e3-b81d-f1d4b1eaf063)
![two-barbarians-673963326](https://github.com/axx0/Civ2-clone/assets/21183395/6bacb3af-4a27-45f1-a5df-c6812cf7bd09)
![barbarians-436644603](https://github.com/axx0/Civ2-clone/assets/21183395/39feede8-27f9-4802-bd53-3f2ef6b42726)
![a--wooden-roman-trebuchet-868855323](https://github.com/axx0/Civ2-clone/assets/21183395/08de4c14-33b9-4dfb-ad01-e9ef450c40b1)
![two-roman-phalanx-soldiers-526099904](https://github.com/axx0/Civ2-clone/assets/21183395/a39f3ef8-de1e-4261-813d-f7364b8580cf)
![american-cavalry-in-blue-uniforms-without-flags-426516538](https://github.com/axx0/Civ2-clone/assets/21183395/4a5b288c-1411-4a0a-8c68-3a6ebacd5c49)
![cavalry-632868070](https://github.com/axx0/Civ2-clone/assets/21183395/14ab3dd0-de55-4cb9-9530-f58149570183)
![two-modern-howitzers-one-firing-699579477](https://github.com/axx0/Civ2-clone/assets/21183395/0859cd09-5d1f-4d5c-bae0-ddd9cb22e180)
![two-world-war-2-cannons-one-firing-722742014](https://github.com/axx0/Civ2-clone/assets/21183395/bb4795de-0058-4b5f-bf43-6d53b9dfa00c)
![two-panzers-one-firing](https://github.com/axx0/Civ2-clone/assets/21183395/9f43a8c3-e676-49b2-9901-143724b9ee84)
![two-panzers-one-firing-872530603](https://github.com/axx0/Civ2-clone/assets/21183395/98235e29-37ab-4a58-a42a-224d4a9e4695)
![two-alpine-troops-in-snow-camoflauge-carrying-ak-47s-standing-on-a-rocky-mountain-peak-472956926](https://github.com/axx0/Civ2-clone/assets/21183395/a3458d32-65a0-4ba5-b2b1-8c5bb4eecf26)
![two-riflemen-wearing-blue-uniforms-in-the-late-1800s-one-firing-his-rifle-623389968](https://github.com/axx0/Civ2-clone/assets/21183395/2f7610fb-bdb4-406b-84dd-c4da67751e0f)
![a-diplomat-with-a-top-hat-from-the-1800s-521461941](https://github.com/axx0/Civ2-clone/assets/21183395/8b2179b6-098b-411f-9a03-148d4b9d8320)
![a-sexy-female-spy-wearing-all-black-with-sunglasses-45855650](https://github.com/axx0/Civ2-clone/assets/21183395/16c30058-1c82-4999-8544-84f255548d84)
![two-mounted-dragoons-one-firing-a-musket-367586089](https://github.com/axx0/Civ2-clone/assets/21183395/5600ab0b-33d2-4f30-88f6-adea582bea40)
![two-napolean-era-cannons-one-firing-8397214](https://github.com/axx0/Civ2-clone/assets/21183395/b9545b15-2884-4877-a02f-a1af0f5c7361)
![two-b-17-bombers-flying-in-formation-from-above-368759508](https://github.com/axx0/Civ2-clone/assets/21183395/c67a2066-6714-43a4-916f-150d0daca10a)
![two-p-71-black-widows-flying-in-formation-from-above-646645186](https://github.com/axx0/Civ2-clone/assets/21183395/8454ec36-772b-47f8-ad46-3c26555def91)
![two-stealth-fighters-flying-in-formation-from-above-819390806](https://github.com/axx0/Civ2-clone/assets/21183395/d8ab2418-8ab4-4b22-b055-def185339828)
![two-stealth-bombers-flying-in-formation-from-above-103805642](https://github.com/axx0/Civ2-clone/assets/21183395/b8298829-d855-498f-8010-fef8cf1701f2)
![two-neanderthals-thrusting-spears-in-unison-towards-towards-the-camera-256148996 (1)](https://github.com/axx0/Civ2-clone/assets/21183395/74cc96ee-27dd-47d6-9a1c-87654a9ad845)
![two-musketeers-both-facing-the-camera-one-standing-and-firing-one-taking-a-knee-and-loading-their--256148996](https://github.com/axx0/Civ2-clone/assets/21183395/9bd21816-4141-4366-8233-a177aa0363b3)
![two-world-one-era-machine-gunners-facing-the-camera-on-a-solid-green-background-256148996](https://github.com/axx0/Civ2-clone/assets/21183395/da9fcb17-57c5-4114-a904-4a15829fabb0)
![two-musketeers-both-facing-the-camera-one-standing-and-firing-one-taking-a-knee-and-loading-their-](https://github.com/axx0/Civ2-clone/assets/21183395/6ab12fee-30aa-4f73-8407-6999825655bb)
![two-neanderthals-thrusting-spears-in-unison-towards-towards-the-camera](https://github.com/axx0/Civ2-clone/assets/21183395/a86974e9-9804-4fba-a7b3-aad88b45ad8e)
![two-neanderthals-thrusting-spears-in-unison-towards-towards-the-camera-256148996](https://github.com/axx0/Civ2-clone/assets/21183395/7b19d13a-e8d9-4ea7-998d-96d22b5f5284)

**Reply by reubene** [reply]
You probably should fork the repo, make the changes you want and then make a PR so we can incorporate them into the main app when needed. 

Also the engine currently assumes the tiles are the same width as the units are you planning on creating tile images as well?

**Reply by axx0** [reply]
I'll make those folders. Do you need me to put this art into them or will you commit?
PNGs will work. But there's currently an issue with transparency (setting alpha) with pngs in raylib.

**Reply by crhy** [reply]
Yes, I'll create all image assets required.  At scale 1024x1024.  Units will have a transparent background, the rest will be solid.  The Tech Advancement art seems to be in a rectangular format, and I think that looks good (although obviously far too low-res in the original LOL).  Can you give me an upscale dimension you think they should be in?

**Reply by crhy** [reply]
Hmmm.. I still can't seem to create folders or upload files.  I'm kind of a github n00b, unfortunately.

**Reply by crhy** [reply]
Do you guys have an IRC chat or somewhere we can communicate slightly more real time, at least to get this process going initially?  LOL  I'm messing up more than I'm fixing I think.  *blush*

**Reply by axx0** [reply]
You probably can't create a folder. I'll create it for you, then you can upload the files and make a pull request.

**Reply by crhy** [reply]
Wait, where do I upload the files?  I've tried a couple things, but admit to being a n00b.

**Reply by crhy** [reply]
I've been working on terrain all day.  It's not perfect, but a good start.
![Uploading looking-down-on-a-densely-packed-clump-of-trees-on-a-solitary-grassy-hill-843333125.png…]()

**Reply by crhy** [reply]
![Uploading looking-down-on-a-densely-packed-clump-of-trees-on-a-solitary-grassy-hill-843333125.png…]()

**Comment by crhy** [comment]
See?  That function doesn't work.

**Comment by crhy** [comment]
This image alone could start a whole new video game franchise: https://files.catbox.moe/wli7gb.png

**Comment by crhy** [comment]
early terrain research: https://files.catbox.moe/nzfwsu.png

**Comment by crhy** [comment]
Just take an isographic diamond out of the upper middle....  would look awesome.

**Comment by crhy** [comment]
Check out this mine:  https://files.catbox.moe/9a2yo5.png

**Reply by crhy** [reply]
OK guys.  Let me know when there's a folder to upload to.  I have some concrete ideas on how to implement the new textures and units.  Basically we can just grab the isometric diamond the current game uses from the upscaled/new images and it should look really good if we fuzz the edges a little during render (every other pixel with adjacent tile).

**Reply by crhy** [reply]
OK.  For now I just dumped them all in the Civ2Gold folder which was mostly empty, and where I'm pulling the data on which files we need anyway.

**Reply by crhy** [reply]
Of course, this is an early rough draft, to see if we can even get them in game, etc..  Will clean them up and rename them and all based on whatever conventions you guys want as needed.  Cheers!  Happy Aloha Friday from Hawaii!

**Reply by axx0** [reply]
ok I've made a folder inside raylibui. I had to put some txt files inside as github won't add empty folders.
Of course we will need to make some logic how the game will read this custom art, as everything we're currently doing is related to the old game and that will take some time.

**Reply by crhy** [reply]
OK, all the first drafts are in Civ2Gold already.  You can move them over if you want.  LOL  I have concrete ideas for version 2.0 of most of them.  Not the spy one though.  It's perfect already.  A few others too actually.

**Reply by crhy** [reply]
PS If you can get the spy in game and I can play it I'll venmo you or send you some BTC.  <3

**Reply by axx0** [reply]
Ok you need to make a PR so I can merge the changes into the project.

**Comment by crhy** [comment]
I think i did that? I don't really know what I'm doing. 😅😅😅 I should
make a telegram channel or discord or something....

**Reply by axx0** [reply]
I use github desktop for that stuff. Anyway just look up online how to do a PR, it shouldn't be too difficult.

**Comment by crhy** [comment]
https://t.me/fossciv2

**Comment by crhy** [comment]
Before I do too much more work on texture/asset replacement, it would be great to see what it looks like in game.  Even a fake screenshot would be useful.

**Comment by crhy** [comment]
By the way I did this, but I have no idea what to do next.  LOL  N000000b alert.

https://github.com/axx0/Civ2-clone/pulls

**Comment by crhy** [comment]
So I also have most of the Wonders and Tech Tree finished.  Though currently they all need to be relabeled.  https://github.com/crhy/Civ2-clone/tree/master/Civ2Gold

**Reply by crhy** [reply]
Also the City Improvements, totally mislabelled.

**Reply by crhy** [reply]
Feel like I've got a good first working draft of most of the stuff now.....  Got to label it all though....  Github is hard to manage.

**Comment by crhy** [comment]
Hey guys!  I'm updating my branch with renamed artwork for all Advances, Units, City Improvements, and Terrain.  We'll still need art for the cities themselves as they grow, and in the different styles.  Unless we want to get rid of the different styles.  I always thought that was pretty pointless.  I've updated my branch with the advances already.  Working on renaming units, improvements, and terrain now!

https://github.com/crhy/Civ2-clone/tree/master/RaylibUI/FOSS%20art

**Reply by crhy** [reply]
City Improvements Uploaded too now.

**Reply by crhy** [reply]
Wonders Uploaded also.

**Reply by crhy** [reply]
Units now uploaded.

**Reply by crhy** [reply]
Terrain Uploaded as well, should Resources be their own separate folder?  I've made a lot of progress on those as well.

**Comment by crhy** [comment]
I created the first pull request, but it's an absolute mess.  LOL

**Reply by axx0** [reply]
That's a gigabyte of assets. I can't even download that because I'm using dropbox for this repository and I'm near the limit.

Keep in mind that these pictures need to be shown in the game. Currently icons for wonders are tens of pixels wide. This is way too big, it would require redesigning all our dialogs to show this. I suggest you either make the pictures smaller to a more manageable size or upload the assets somewhere online so we can access them.

**Comment by crhy** [comment]
The point was to have the assets so they are a little bit future-proof, and
we can zoom way in and they won't lose quality.

Understanding that in the game even in 4K they are probably going to be
less than 200 pixels across generally.

I haven't done any compression work though. Right now they're all just
PNGs, it seems like JPEG is more efficient. Should I convert them all to
JPEGs and re-upload?

How do I remove stuff from GitHub? It's so annoying that I can't figure out
how to do that on my own.

rhY

**Reply by axx0** [reply]
Don't PNGs themselves have options for compression? Anyway if you can't compress them I suggest you store them on your HD until we get to the point where can use them. Right now our hands are full with more basic game stuff.

> How do I remove stuff from GitHub? It's so annoying that I can't figure out
how to do that on my own.

When you do a commit, it reflects changes that you made on your local repository. So if you remove stuff from there and do a commit, it should be removed on your repository.

**Comment by crhy** [comment]
Yes I definitely have them stored on my hard drive. No worries. I will
experiment with compression later this week. I'm on tour right now with the
LeperKhanz. https://rumble.com/v4jkxpx-leperkhanz-cottage-in-the-grove.html

I'm just doing everything straight from the GitHub web interface. I'm not
keeping anything local anywhere. Other than the images that I'm editing of
course. But I don't keep a local repo. I'm on a laptop on the road so....

Cheers guys!  Pumped on the whole project!
rhY

On Fri, Mar 15, 2024 at 8:27 AM axx ***@***.***> wrote:

> Don't PNGs themselves have options for compression? Anyway if you can't
> compress them I suggest you store them on your HD until we get to the point
> where can use them. Right now our hands are full with more basic game stuff.
>
> How do I remove stuff from GitHub? It's so annoying that I can't figure out
> how to do that on my own.
>
> When you do a commit, it reflects changes that you made on your local
> repository. So if you remove stuff from there and do a commit, it should be
> removed on your repository.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/axx0/Civ2-clone/discussions/102#discussioncomment-8806195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFBTXI3LULDDYII2JFRJC6TYYM4RXAVCNFSM6AAAAABDISRHXOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DQMBWGE4TK>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
- Cowboy Crooner, Violinist, Composer, Linux Guy, World's Best Dad, AirShip
Designer

rhymusic.com
OpenAirShips.com
ThorntonTechs.com
CoinKitty.org
SuperMox.me
https://leperkhanz.weebly.com/ <http://leperkhanz.weebly.com/>

**Comment by crhy** [comment]
If I export them as jpg at 76% compression 1.5mb becomes 225kb.  Is that an
acceptable filesize?

Let me know if that works....

rhY

On Sat, Mar 16, 2024 at 9:38 AM rhy o'drinnan ***@***.***> wrote:

> Yes I definitely have them stored on my hard drive. No worries. I will
> experiment with compression later this week. I'm on tour right now with the
> LeperKhanz.
> https://rumble.com/v4jkxpx-leperkhanz-cottage-in-the-grove.html
>
> I'm just doing everything straight from the GitHub web interface. I'm not
> keeping anything local anywhere. Other than the images that I'm editing of
> course. But I don't keep a local repo. I'm on a laptop on the road so....
>
> Cheers guys!  Pumped on the whole project!
> rhY
>
> On Fri, Mar 15, 2024 at 8:27 AM axx ***@***.***> wrote:
>
>> Don't PNGs themselves have options for compression? Anyway if you can't
>> compress them I suggest you store them on your HD until we get to the point
>> where can use them. Right now our hands are full with more basic game stuff.
>>
>> How do I remove stuff from GitHub? It's so annoying that I can't figure
>> out
>> how to do that on my own.
>>
>> When you do a commit, it reflects changes that you made on your local
>> repository. So if you remove stuff from there and do a commit, it should be
>> removed on your repository.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/axx0/Civ2-clone/discussions/102#discussioncomment-8806195>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AFBTXI3LULDDYII2JFRJC6TYYM4RXAVCNFSM6AAAAABDISRHXOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DQMBWGE4TK>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
> - Cowboy Crooner, Violinist, Composer, Linux Guy, World's Best Dad,
> AirShip Designer
>
> rhymusic.com
> OpenAirShips.com
> ThorntonTechs.com
> CoinKitty.org
> SuperMox.me
> https://leperkhanz.weebly.com/ <http://leperkhanz.weebly.com/>
>


-- 
- Cowboy Crooner, Violinist, Composer, Linux Guy, World's Best Dad, AirShip
Designer

rhymusic.com
OpenAirShips.com
ThorntonTechs.com
CoinKitty.org
SuperMox.me
https://leperkhanz.weebly.com/ <http://leperkhanz.weebly.com/>

**Reply by axx0** [reply]
Yes much better.

**Reply by crhy** [reply]
OK, I'll redo all of them, and upload shortly.  Any idea on how to delete the ones I uploaded already?  Or should I make a different folder, or.... ?

**Reply by axx0** [reply]
If you delete the files on your drive and do the commit the files should be deleted on the repo also (I think).

**Comment by crhy** [comment]
I only use the web interface... 😭😭😭

**Reply by axx0** [reply]
In that case just do a new pull request with new files. I'll remove the previous two.

**Reply by crhy** [reply]
OK, so I'm finally back to civilization (reality with a house and a computer).  Just so I'm not doing twice the work, what are our format goals?  The troops are transparent cut outs, so that precludes using jpg.  I can do gif or png, but the default png compression seems quite large file size wise.  The other images are all square and I can do jpg and make them super small and still high quality.  What is the size goal per image, given that I've already got most images in 1024x1024 format?  200k?  300k?  Thanks in advance.

**Reply by axx0** [reply]
Raylib can read gifs and pngs, but I haven't tried using jpgs.
I've noticed some issues with pngs and transparency however (if you make a new png it works ok, but if you convert a say bmp to png it won't show the transparent parts as actually transparent).
I'm not sure what the best size is, but the smaller the better. Also keep in mind github has limits to repository sizes.

**Comment by crhy** [comment]
OK, I really don't know how to use github.  At all.  But I deleted the large assets and replaced them with much smaller ones.  From 400mb down to 80mb roughly.  I used jpg and gif instead of png.  I left them the same resolution so we can stay future proof with higher resolution displays or VR or whatever.  They look great, and are MUCH smaller now.  Uploading files is a drag.  A lot of times the processing of files fails and I have to start over.

  I've updated my branch:  https://github.com/crhy/Civ2-clone

I'm not sure how to make that into a pull request?  I somehow was able to close the two previous pulls.  Anyway, let me know if I need to do something else, or if there are some assets missing or anything!  Cheers guys!  <3

**Comment by crhy** [comment]
I uploaded the better compressed files today. Over four times smaller than
the originals. Let me know if that works out!

**Comment by axx0** [comment]
Keep the files on your repo, we'll merge them here eventually when we can actually use them. There's loads of other stuff to do before that can happen.

**Comment by crhy** [comment]
Ok, sounds good. Thanks! 😍😍😍

**Comment by crhy** [comment]
Just FYI I upgraded my fork with the FOSS art:

https://github.com/crhy/Civ2-clone/blob/master/README.md

Probably a lot to clean up still. 

Also I'm happy to retype our edit the text dialogs as well.... Though I haven't started that at all.

**Reply by reubene** [reply]
Yeah, when we get to incorporating your work from this branch we're going to need a full set of menus, dialogs and labels.  They don't have to bee 100% the same as the default but they need to be in a consistent format.

**Comment by crhy** [comment]
I can probably get started on that this week. Plenty of valid info on the
civopedia. Traveling in California right now. Messaged in the chat on
telegram too.

- Cowboy Crooner, Violinist, Composer, Linux Guy, World's Best Dad, AirShip
Designer

rhymusic.com
OpenAirShips.com
ThorntonTechs.com
CoinKitty.org
SuperMox.me
https://leperkhanz.weebly.com/ <http://leperkhanz.weebly.com/>

**Comment by crhy** [comment]
Hey guys!  Got done with the LeperKhanz tour, and am ready to get back to "work".  What's next that I can add?  And my branch is ready to pull with the graphics we have done so far.  I'll work on Leader Graphics right now.

**Comment by crhy** [comment]
![mao](https://github.com/user-attachments/assets/241f355e-74c9-45ae-b353-be595f5b8cfc)

Here's one of the leaders.  LMK if I'm going in the right direction or if we even need these.  Also, are we married to the leaders the original chose?  IMHO there are better leaders for some of the nations.  Definitely Germany, India, England, Ireland.  For example.

My favorite Celt leader:
![MichaelCollins](https://github.com/user-attachments/assets/7db930e6-d38b-45e6-b4e3-2e928ef516f9)

**Comment by crhy** [comment]
Also, it would be cool if the diplomacy screen just blurred the map in the background, but otherwise functioned the same as before.

![Diplomacy](https://github.com/user-attachments/assets/4e20e6c8-10b8-427a-88a9-0bdb805750ee)

**Comment by crhy** [comment]
Also I'm playing around with the textures I made and seeing what they'll look like in a map.  This is extremely rough draft.  I hate the plains texture now, and the ocean needs to be a lot more blue.  The salmon and whale are cool so far though!

![terraincollage](https://github.com/user-attachments/assets/6ae98bbe-2656-4b5d-ad69-9146ca6341ac)

**Reply by crhy** [reply]
Also, I'm wondering how the original engine made the coast line tiles.  Maybe that needs to be seperate specific art?  The original game is so much better than you realize once you start trying to re-engineer it!

**Reply by reubene** [reply]
There are special coast tile sections each 1/4 the size of a tile. one for each combination of land and water at a vertex

**Reply by crhy** [reply]
I should look at the original art and use that for a template.  On it.

Posted this thread too:  https://forums.civfanatics.com/threads/upgrading-the-original-terrain-artwork.697586/

**Comment by crhy** [comment]
OK, I finished editing and compressing the leaders.  Here's the link in my fork:  https://github.com/crhy/FOSS-Civ2-Clone/tree/master/RaylibUI/FOSS%20art/Leaders

**Comment by crhy** [comment]
Just FYI cleaned up and renamed my fork repo today.  Found some duplicate files to delete, and synced with the main branch.

**Comment by crhy** [comment]
These guys have upgraded Terrain already for Civ 2:  https://forums.civfanatics.com/threads/better-terrain-graphics-for-mge-from-eivinds-colonialism-scenario.667221/

Let's see if they'll let us include it in AxCiv

**Reply by crhy** [reply]
Here's their response: I built the mod using other peoples work from decades ago and only touched up a few things here and there so not really up to me. However as long the project stays opensource and the assets aren't used in a commercial endeavor I'm sure the old artists won't mind.

**Reply by crhy** [reply]
So that sounds like a go!
